src/database/migrations/20231211041333_CreateApiTable.ts

```diff
@@ -0,0 +1,32 @@
+import { Knex } from 'knex';
+
+import { ApiRole } from '../../api/entities/ApiRole';
+import { Participant } from '../../api/entities/Participant';
+import { getSiteList } from '../../api/services/adminServiceClient';
+import { SiteDTO } from '../../api/services/adminServiceHelpers';
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable('apiRoles', (table) => {
+    table.increments('id').primary();
+    table.string('roleName', 100).notNullable();
+  });
+
+  // Add default roles
+  const defaultRoles = ['MAPPER', 'GENERATOR', 'ID_READER', 'SHARER', 'OPTOUT'];
+  await knex('apiRoles').insert(
+    defaultRoles.map((role) => {
+      return { roleName: role };
+    })
+  );
+
+  await knex.schema.createTable('participantsToRoles', (table) => {
+    table.integer('participantId').references('participants.id').onDelete('CASCADE');
+    table.integer('apiRoleId').references('id').inTable('apiRoles').onDelete('CASCADE');
```

Comment:
I don't think we want the cascade here: if someone tries to delete a role that's in use, I'd prefer to have the delete fail.

The delete on participants is OK - I don't imagine we'll delete participants often (or ever), but if we do, removing the linking table entry makes sense.

Reply:
Cool, yeah that makes sense.